draft-ietf-quic-applicability.md

```diff
@@ -454,6 +454,21 @@ Some deadlocking scenarios might be resolved by cancelling affected streams with
 STOP_SENDING or RESET_STREAM.  Cancelling some streams results in the connection
 being terminated in some protocols.
 
+# Stream Limit Commitments
+
+QUIC endpoints can manage the cumulative maximum number of streams they would
+allow to be opened using the MAX_STREAMS frame, but there is no mechanism to
+reduce the value. An application that uses QUIC might commit to a number of
+openable streams but require the connection to be closed (for example, a
+scheduled maintenance period). Depending on how an application uses QUIC
+streams, abrupt closure of actively used streams may be undesireable or
+detrimental. In contrast, waiting for an endpoint to exhaust the advertised
+limit may not suit application or operational needs. Applications using QUIC can
```

Comment:
I think I miss some connection here: why should an endpoint ever wait until all stream are exhausted...?

Reply:
So imagine an HTTP/3 server that advertises that up to 10 bidirectional client streams streams can be created. The client has created 2 streams for requests,and the responses are large payloads. 

The server gets scheduled for an upgrade and the operator wants to drain the connection gracefully. It is happy to let the large downloads conclude but it wants new requests to get routed to a new machine. Any form of connection closure at this stage risks affecting the in-flight response. Connection closure when all streams are exhausted is a much safer option.

The client believes it can create 8 more streams and it is sub optimal to create a new connection while the current one appears to be able to handle what is needed. It could also be bad to open concurrent connections opportunistically in anticipation of a server having to gracefully close. The server could just wait for the client to create those 8 streams and the instantly reject them with STOP_SENDING, RESET_STREAM, or an HTTP response. That's a very chatty process, and might delay the client from being able to fulfill its duties in a timely manner. 

HTTP/3 GOAWAY is a graceful close signal that permits server to tell the client that it won't accept new requests, but it doesn't interrupt active streams. A client that receives GOAWAY could decide to open a new connection in parallel to the active transfers completing.

Reply:
I understand why you need/want GOAWAY at the application.I'm just think that waiting until stream are exhausted is never really an option because as soon as one stream is closed the client can immediately open another one. I think the two options you have is you either wait until the connection is done entirely, or you have an application layer mechanism that indicates that no new streams should be created.

Reply:
The client can only open as many as MAX_STREAMs. So if a server wants to drain a connection without a GOWAY, it can stop granting stream credits (this is what I mean by exhaustion) and eventually the client will run out of things to do. From the other guidance in the spec, a client with nothing to do should eventually give up either via idle timeout or explicit close.

Reply:
Okay, you are right. I was misremembering how MAX_STREAM works (would maybe make sense to add a sentence). So basically what you are saying is that if the server wants to close the connection but doesn't want to interrupt on-going activities, the only thing it can do is to not issue any new stream credits and wait until the client is done because it easier has nothing to send anymore or reaches the stream limit (if you don't have an graceful close on application level). I think this can be worded simpler than taking about "exhausting streams" :-)

Reply:
I updated the text to address your comments,  PTAL